src/main/java/com/wavefront/opentracing/WavefrontTracer.java

```diff
@@ -175,38 +166,30 @@ public Builder withGlobalMultiValuedTags(Map<String, Collection<String>> tags) {
     }
 
     /**
-     * Set the reporter for this tracer.
+     * Apply ApplicationTags as global span tags.
      *
-     * @param reporter the reporter for this tracer
+     * @param applicationTags Metadata about your application.
      * @return {@code this}
      */
-    public Builder withReporter(Reporter reporter) {
-      this.reporter = reporter;
+    public Builder withApplicationTags(ApplicationTags applicationTags) {
+      withGlobalTag(APPLICATION_TAG_KEY, applicationTags.getApplication());
+      withGlobalTag(SERVICE_TAG_KEY, applicationTags.getService());
+      withGlobalTag(CLUSTER_TAG_KEY,
+          applicationTags.getCluster() == null ? NULL_TAG_VAL : applicationTags.getCluster());
+      withGlobalTag(SHARD_TAG_KEY,
+          applicationTags.getShard() == null ? NULL_TAG_VAL : applicationTags.getShard());
       return this;
     }
 
+
     /**
      * Builds and returns the WavefrontTracer instance based on the provided configuration.
      *
      * @return a {@link WavefrontTracer}
      */
-    public WavefrontTracer build() {
-      if (source == null || source.isEmpty()) {
-        source = getDefaultSource();
-      }
-      if (reporter == null) {
-        reporter = new ConsoleReporter(source);
-      }
+    public WavefrontTracer build(Reporter reporter) {
```

Comment:
Can you move the reporter to Builder() constructor? (to be consistent with builders in other sdk)